wrapper/CSharp/wolfTPM.cs

```diff
@@ -0,0 +1,833 @@
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            ret = wolfTPM2_LoadKey(dev, blob, handle);
+            if (ret != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_LoadKey() failed.");
+            }
+        }
+
+        if (parent_key != IntPtr.Zero) {
+            if (wolfTPM2_CleanupKey(parent_key) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_CleanupKey() failed.");
+            }
+        }
+
+        if (blob != IntPtr.Zero) {
+            if (wolfTPM2_CleanupKeyBlob(blob) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_CleanupKeyBlob() failed.");
+            }
+        }
+
+        if (dev != IntPtr.Zero) {
+            if (wolfTPM2_SimpleCleanup(dev) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_SimpleCleanup() failed.");
+            }
+        }
+
+        Console.WriteLine("Leaving TryLoadGeneratedKey(" + algorithm +
+            ") with ret=" + ret);
+        Console.WriteLine("");
+    }
+
+    public static void TryLoadRSAPublicKey() {
+        int ret = TPM_RC_SUCCESS;
+        IntPtr dev = IntPtr.Zero;
+        IntPtr pub_key = IntPtr.Zero;
+        byte[] pub_buffer;
+        int exp = 0x10001;
+
+        pub_buffer = File.ReadAllBytes("rsa_public.bin");
+        PrintByteArray(pub_buffer);
+
+        dev = wolfTPM2_SimpleInit();
+        if (dev == IntPtr.Zero) {
+            Console.WriteLine("wolfTPM2_SimpleInit() failed.");
+            ret = -1;
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            pub_key = wolfTPM2_GetEmptyKey();
+            if (pub_key == IntPtr.Zero) {
+                Console.WriteLine("wolfTPM2_GetEmptyKey() failed.");
+                ret = -1;
+            }
+        }
+
+        if (ret == TPM_RC_SUCCESS) {
+            ret = wolfTPM2_LoadRsaPublicKey(dev, pub_key, pub_buffer,
+                                            pub_buffer.Length, exp);
+            if (ret != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_LoadRsaPublicKey() failed.");
+            }
+        }
+
+        if (pub_key != IntPtr.Zero) {
+            if (wolfTPM2_CleanupKey(pub_key) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_CleanupKey(pub_key) failed.");
+            }
+        }
+
+        if (dev != IntPtr.Zero) {
+            if (wolfTPM2_SimpleCleanup(dev) != TPM_RC_SUCCESS) {
+                Console.WriteLine("wolfTPM2_SimpleCleanup() failed.");
+            }
+        }
+
+        Console.WriteLine("Leaving TryLoadRSAPublicKey() with ret=" + ret);
+        Console.WriteLine("");
+    }
+
+    public static void TryLoadRSAPrivateKey() {
+        int ret = TPM_RC_SUCCESS;
+        IntPtr dev = IntPtr.Zero;
+        IntPtr parent_key = IntPtr.Zero;
+        IntPtr priv_key = IntPtr.Zero;
+        byte[] pub_buffer;
+        byte[] priv_buffer;
+        int exp = 0x10001;
+
+        pub_buffer = File.ReadAllBytes("rsa_public.bin");
```

Comment:
I don't think these file are portable or should be committed. Instead you should be creating a key from the TPM and saving these for the test.

Reply:
These files are based on `kRsaKeyPubModulus` and `kRsaKeyPrivQ` in `examples/tpm_test.h`: 
```
/* from wolfSSL ./certs/client-key.der */
static const byte kRsaKeyPubModulus[] = {
    0xC3, 0x03, 0xD1, 0x2B, 0xFE, 0x39, 0xA4, 0x32, 0x45, 0x3B, 0x53, 0xC8, 0x84,
    0x2B, 0x2A, 0x7C, 0x74, 0x9A, 0xBD, 0xAA, 0x2A, 0x52, 0x07, 0x47, 0xD6, 0xA6,
    0x36, 0xB2, 0x07, 0x32, 0x8E, 0xD0, 0xBA, 0x69, 0x7B, 0xC6, 0xC3, 0x44, 0x9E,
    0xD4, 0x81, 0x48, 0xFD, 0x2D, 0x68, 0xA2, 0x8B, 0x67, 0xBB, 0xA1, 0x75, 0xC8,
    0x36, 0x2C, 0x4A, 0xD2, 0x1B, 0xF7, 0x8B, 0xBA, 0xCF, 0x0D, 0xF9, 0xEF, 0xEC,
    0xF1, 0x81, 0x1E, 0x7B, 0x9B, 0x03, 0x47, 0x9A, 0xBF, 0x65, 0xCC, 0x7F, 0x65,
    0x24, 0x69, 0xA6, 0xE8, 0x14, 0x89, 0x5B, 0xE4, 0x34, 0xF7, 0xC5, 0xB0, 0x14,
    0x93, 0xF5, 0x67, 0x7B, 0x3A, 0x7A, 0x78, 0xE1, 0x01, 0x56, 0x56, 0x91, 0xA6,
    0x13, 0x42, 0x8D, 0xD2, 0x3C, 0x40, 0x9C, 0x4C, 0xEF, 0xD1, 0x86, 0xDF, 0x37,
    0x51, 0x1B, 0x0C, 0xA1, 0x3B, 0xF5, 0xF1, 0xA3, 0x4A, 0x35, 0xE4, 0xE1, 0xCE,
    0x96, 0xDF, 0x1B, 0x7E, 0xBF, 0x4E, 0x97, 0xD0, 0x10, 0xE8, 0xA8, 0x08, 0x30,
    0x81, 0xAF, 0x20, 0x0B, 0x43, 0x14, 0xC5, 0x74, 0x67, 0xB4, 0x32, 0x82, 0x6F,
    0x8D, 0x86, 0xC2, 0x88, 0x40, 0x99, 0x36, 0x83, 0xBA, 0x1E, 0x40, 0x72, 0x22,
    0x17, 0xD7, 0x52, 0x65, 0x24, 0x73, 0xB0, 0xCE, 0xEF, 0x19, 0xCD, 0xAE, 0xFF,
    0x78, 0x6C, 0x7B, 0xC0, 0x12, 0x03, 0xD4, 0x4E, 0x72, 0x0D, 0x50, 0x6D, 0x3B,
    0xA3, 0x3B, 0xA3, 0x99, 0x5E, 0x9D, 0xC8, 0xD9, 0x0C, 0x85, 0xB3, 0xD9, 0x8A,
    0xD9, 0x54, 0x26, 0xDB, 0x6D, 0xFA, 0xAC, 0xBB, 0xFF, 0x25, 0x4C, 0xC4, 0xD1,
    0x79, 0xF4, 0x71, 0xD3, 0x86, 0x40, 0x18, 0x13, 0xB0, 0x63, 0xB5, 0x72, 0x4E,
    0x30, 0xC4, 0x97, 0x84, 0x86, 0x2D, 0x56, 0x2F, 0xD7, 0x15, 0xF7, 0x7F, 0xC0,
    0xAE, 0xF5, 0xFC, 0x5B, 0xE5, 0xFB, 0xA1, 0xBA, 0xD3
};

static const word32 kRsaKeyPubExponent = 0x10001; /* {0x01, 0x00, 0x01} */

static const byte kRsaKeyPrivQ[] = {
    0xD5, 0x38, 0x1B, 0xC3, 0x8F, 0xC5, 0x93, 0x0C, 0x47, 0x0B, 0x6F, 0x35, 0x92,
    0xC5, 0xB0, 0x8D, 0x46, 0xC8, 0x92, 0x18, 0x8F, 0xF5, 0x80, 0x0A, 0xF7, 0xEF,
    0xA1, 0xFE, 0x80, 0xB9, 0xB5, 0x2A, 0xBA, 0xCA, 0x18, 0xB0, 0x5D, 0xA5, 0x07,
    0xD0, 0x93, 0x8D, 0xD8, 0x9C, 0x04, 0x1C, 0xD4, 0x62, 0x8E, 0xA6, 0x26, 0x81,
    0x01, 0xFF, 0xCE, 0x8A, 0x2A, 0x63, 0x34, 0x35, 0x40, 0xAA, 0x6D, 0x80, 0xDE,
    0x89, 0x23, 0x6A, 0x57, 0x4D, 0x9E, 0x6E, 0xAD, 0x93, 0x4E, 0x56, 0x90, 0x0B,
    0x6D, 0x9D, 0x73, 0x8B, 0x0C, 0xAE, 0x27, 0x3D, 0xDE, 0x4E, 0xF0, 0xAA, 0xC5,
    0x6C, 0x78, 0x67, 0x6C, 0x94, 0x52, 0x9C, 0x37, 0x67, 0x6C, 0x2D, 0xEF, 0xBB,
    0xAF, 0xDF, 0xA6, 0x90, 0x3C, 0xC4, 0x47, 0xCF, 0x8D, 0x96, 0x9E, 0x98, 0xA9,
    0xB4, 0x9F, 0xC5, 0xA6, 0x50, 0xDC, 0xB3, 0xF0, 0xFB, 0x74, 0x17
};

```
I do create and export and import .  However, this is for loading an external key which is a different use case.